tools/extract_scripts.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+# these two pages have some relevant information:
+# https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+# https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+
+import argparse
+import re
+import sys
+from pathlib import Path
+from typing import Any, Dict
+
+import yaml
+
+Job = Dict[str, Any]
+
+windows_labels = {'windows-latest', 'windows-2019'}
+
+
+def get_default_shell(job: Job) -> str:
+    if job['runs-on'] in windows_labels:
+        return 'pwsh'
+    else:
+        return 'bash'
```

Comment:
Nit (I like ternaries for some reason)
```suggestion
    return 'pwsh' if job['runs-on'] in windows_labels else 'bash'
```